README.md

```diff
@@ -297,6 +299,96 @@ and then change its colors, the items missed would fallback to use the default c
 Please also notice that an empty theme is **NOT** supported due to
 [a bug in serde lib](https://github.com/dtolnay/serde-yaml/issues/86).
 
+### Icon Theme
+
+Icon theme can be configured in a fixed location, `$XDG_CONFIG_DIR/lsd/icons.yaml`,
+for example, `~/.config/lsd/icons.yaml` on macOS,
+please check [Config file location](#config-file-location) to make sure where is `$XDG_CONFIG_DIR`.
+
+As the file name indicated, the icon theme file is a `yaml` file.
+
+Check [Icon Theme file content](#icon-theme-file-content) for details.
+
+#### Icon Theme file content
+
+lsd support 3 kinds of icon configuration:
+- filetype
+- name
+- extension
+
+The default icon theme scheme shipped with `lsd` can be check in [icon theme source code](src/theme/icon.rs), we will load the default theme, and overwrite it with user defined parts, here is a example for icon theme.
+
+lsd icon theme support both nerd font and Unicode in the same time, you can use any one to config the theme, this is the default but you can customize this to use glyphs from nerd-font or Unicode emojis in the same time.
+
+nerd font:
+
+```yaml
+name:
+  .trash: 
+  .cargo: 
+  .emacs.d: 
+  a.out: 
+extension:
+  go: 
+  hs: 
+  rs: 
+filetype:
+  dir: 
+  file: 
+  pipe: 
+  socket: 
+  executable: 
+  symlink-dir: 
+  symlink-file: 
+  device-char: 
+  device-block: ﰩ
+  special: 
+```
+
+Unicode:
+
+```yaml
+name:
+  .trash: 🗑
+extension:
+  rs: 🦀
+filetype:
+  dir: 📂
+  file: 📄
+  pipe: 📩
+```
+
+mixed:
+
+```yaml
+name:
+  .trash: 
+  .cargo: 
+  .emacs.d: 
+  a.out: 
+extension:
+  go: 
+  hs: 
+  rs: 🦀
+filetype:
+  dir: 📂
+  file: 📄
+  pipe: 📩
+  dir: 
+  file: 
+  pipe: 
+  socket: 
+  executable: 
+  symlink-dir: 
+  symlink-file: 
+  device-char: 
+  device-block: ﰩ
+  special: 
+```
+
+When creating a theme for `lsd`, you can specify any part of the default theme,
+and then change its icons, the items missed would fallback to use the default icons.
```

Comment:
This doc block felt a bit obtuse and thought I would rework it. What do you think?

```suggestion
`lsd` support 3 kinds of icon overrides, by `name`, by `filetype` and by `extension`.
The final set of icons used will be a combination of what is shipped with in `lsd` with overrides from config applied on top of it.
*You can find the default set of icons [here](src/theme/icon.rs).*

Both nerd font glyphs and unicode emojis can be used for icons. You can find an example of icons customization below.

```yaml
name:
  .trash: 
  .cargo: 
  .emacs.d: 
  a.out: 
extension:
  go: 
  hs: 
  rs: 🦀
filetype:
  dir: 📂
  file: 📄
  pipe: 📩
  socket: 
  executable: 
  symlink-dir: 
  symlink-file: 
  device-char: 
  device-block: ﰩ
  special: 
```
```

Writing this actually gave me another idea, we could rename the `--icon-type unicode` to `--icon-type simple` and with this we will only use icons defined in `filetype` and not use anything by extension or name. Their configs will also affect in this case. As of now, it is kinda as if those don't do much. I'll add it to the issue, but just thought I would mention it here.